src/ca/wolfram/beta/symath/MathNode.java

```diff
@@ -0,0 +1,25 @@
+package ca.wolfram.beta.symath;
+
+import java.util.List;
+
+public interface MathNode {
+
+    // static MathNode create(ArrayList<MathNode> nodes);
+    // static MathNode create(MathNode ... nodes);
+    NodeType getType();
+
+    List<MathNode> getChildren();
+
+    boolean isConstant();
+
+    double eval();
+
+    /**
+     * Simplifies node & its children
+     *
+     * @return true if result is a constant
```

Comment:
what defines "successfully simplified"? Let's say it is simplified and still has variables, what does knowing that tell us? If it's a constant we can do quite a bit more, and that is when we know if things are truly simplified.